aws/resource_aws_default_subnet.go

```diff
@@ -73,52 +73,56 @@ func resourceAwsDefaultSubnetCreate(d *schema.ResourceData, meta interface{}) er
 	log.Printf("[DEBUG] Reading Default Subnet: %s", req)
 	resp, err := conn.DescribeSubnets(req)
 	if err != nil {
+		d.SetId("vpc-blank")
 		return nil
 	}
 	if len(resp.Subnets) != 1 || resp.Subnets[0] == nil {
+		d.SetId("vpc-blank")
 		return nil
 	}
-	d.SetId(aws.StringValue(resp.Subnets[0].SubnetId))
-
-	log.Printf("[INFO] Deleting subnet: %s", d.Id())
-
-	if err := deleteLingeringLambdaENIs(conn, "subnet-id", d.Id(), d.Timeout(schema.TimeoutDelete)); err != nil {
-		return fmt.Errorf("error deleting Lambda ENIs using subnet (%s): %s", d.Id(), err)
-	}
-
-	req2 := &ec2.DeleteSubnetInput{
-		SubnetId: aws.String(d.Id()),
-	}
-
-	wait := resource.StateChangeConf{
-		Pending:    []string{"pending"},
-		Target:     []string{"destroyed"},
-		Timeout:    d.Timeout(schema.TimeoutDelete),
-		MinTimeout: 1 * time.Second,
-		Refresh: func() (interface{}, string, error) {
-			_, err := conn.DeleteSubnet(req2)
-			if err != nil {
-				if apiErr, ok := err.(awserr.Error); ok {
-					if apiErr.Code() == "DependencyViolation" {
-						// There is some pending operation, so just retry
-						// in a bit.
-						return 42, "pending", nil
+	if len(resp.Subnets) >= 1 || resp.Subnets[0] != nil {
```

Comment:
Should this be an &&?  if we have a length >=1, then the subnets[0] != nil never fire?

It is also a bit strange that on line 79 we are doing len != 1 but we are then checking resp.subnets[0]... if length is 0, would that blow up?